### PR TITLE
[fix][server] review comment parsing with trim path

### PR DESCRIPTION
**Pull request by gyorb**
Fixes the source file paths extracted at the server and
used for review status parsing when the trim path
feature was used.
The server depends on the client to send all the source files
containing review status comments. After extraction the
received source files will be parsed for review comments
to update them in the database.
Because the wrong path was used the server did not found
the source file to parse.
The server did not throw any error if a source file was missing
because it is a valid usecase that the client does not send
all the source files because they are already available in the
database at the server side.

New test was added where the review comments were modified
between the two store commands (one with trim path) and then
the successful review comment updates are checked.

### CODE REVIEW

File: web/server/codechecker_server/api/report_server.py [hunk 1]
```diff
@@ -2756,8 +2756,13 @@ def get_analyzer_name(report):
                 already_added.add(report_path_hash)
 
                 last_report_event = report.bug_path[-1]
+
+                # The original file path is neded here not the trimmed
```
Comment (csordasmarton) [inline web/server/codechecker_server/api/report_server.py:2760]:
```suggestion
                # The original file path is needed here not the trimmed
```

File: web/server/codechecker_server/api/report_server.py [hunk 2]
```diff
@@ -2756,8 +2756,13 @@ def get_analyzer_name(report):
...
+                # because the source files are extracted as the original
+                # filepath.
```
Comment (csordasmarton) [inline web/server/codechecker_server/api/report_server.py:2762]:
```suggestion
                # file path.
```

**Review by csordasmarton — CHANGES_REQUESTED** [review]
Just some typos and please check why sqlite related test cases are failed. Otherwise LGTM!

**Comment by gyorb** [comment]
I'm not sure why the sqlite test failed to cleanup the workspace, I've pushed the changes let's see if it will be successful now/

**Review by csordasmarton — APPROVED** [review]
LGTM!

**Pull request merged**